### PR TITLE
feat: margin open long

### PR DESCRIPTION
**Pull request by timlind**
[x] msg server
[x] cli

### CODE REVIEW

File: x/margin/keeper/keeper.go [hunk 1]
```diff
@@ -102,3 +105,112 @@ func (k Keeper) DestroyMTP(ctx sdk.Context, symbol string, mtpAddress string) er
...
+	   Notice this is NOT a simple hybrid of uniswap model and Thorchain slippaged based model
+	   But a upgraded version that include swap, updating bouding curve (to be inside the old one)
+	   One can think about this as a state jump:
+	*/
+
+	// todo review
+	//receivedAmount := sentAmount.Mul(receivedBalance.Add(receivedLiabilities)).
+	//	Quo(sentAmount.Add(sentBalance).Add(sentLiabilities)).
+	//	Add(sentAmount.Mul(receivedBalance.Add(receivedLiabilities)).
+	//	Mul(sentBalance.Add(sentLiabilities)).Quo(sentAmount.Add(sentBalance).Add(sentLiabilities))).MulUint64(2)
```
Comment (timlind) [inline x/margin/keeper/keeper.go:134]: @AustinoBombino can we get this on the spec as a multi step function using the sdk math functions?

Comment (AustinoBombino) [reply x/margin/keeper/keeper.go:134]: @timlind Nice catch, Just got into this and realized that it also probably needs to be more sophisticated to account for our decimal normalization logic.

Comment (AustinoBombino) [reply x/margin/keeper/keeper.go:134]: @timlind CustodySwap with sdk math functions and decimal normalization is in this PR: https://github.com/Sifchain/Documentation-Private/pull/2

File: x/margin/types/genesis.go [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package types
+
+import sdk "github.com/cosmos/cosmos-sdk/types"
+
+func DefaultGenesis() *GenesisState {
+	return &GenesisState{
+		// TODO review default param values
+		Params: &Params{
+			LeverageMax: sdk.NewUint(0),
+		},
```
Comment (timlind) [inline x/margin/types/genesis.go:10]: @AustinoBombino are there default values we should be using here for the params?

Comment (AustinoBombino) [reply x/margin/types/genesis.go:10]: Yes, this will be something we'll have to tune. But we can start it at 1.

**Pull request merged**